### ISSUE TITLE
something in the federation sender is dropping logcontexts

### ISSUE THREAD

**Issue by richvdh** [matrix-org/synapse#5271]
Lots of this sort of stuff going on:
```
2019-05-28 10:32:06,362 - synapse.util.metrics - 111 - WARNING - - Context has unexpectedly changed from 'federation_transaction_transmission_loop-6779' to 'sentinel'. ('_send_new_transaction')
2019-05-28 10:32:06,362 - synapse.util.logcontext - 279 - WARNING - - Expected logging context federation_transaction_transmission_loop-6779 has been lost
2019-05-28 10:32:06,365 - synapse.util.metrics - 111 - WARNING - - Context has unexpectedly changed from 'federation_transaction_transmission_loop-6721' to 'sentinel'. ('outbound_request')
```

As with all logcontext drops, this is problematic because (a) it means that logs are hard to trace and (b) metrics will be incorrect

**Comment by richvdh** [comment]
seems to have been introduced by #5037. hopefully fixed in #5277

**Comment by richvdh** [comment]
fixed by #5277
